superset-frontend/src/views/components/RightMenu.tsx

```diff
@@ -118,8 +118,8 @@ const RightMenu = ({
     ALLOWED_EXTENSIONS,
     HAS_GSHEETS_INSTALLED,
   } = useSelector<any, ExtentionConfigs>(state => state.common.conf);
-  const [showModal, setShowModal] = useState<boolean>(false);
-  const [engine, setEngine] = useState<string>('');
+  const [showModal, setShowModal] = React.useState<boolean>(false);
```

Comment:
What is the reason for using this versus importing in `useState` and invoking it normally?

Reply:
So we can `spyOn` during our tests, e.g. `const useStateMock = jest.spyOn(React, 'useState');` https://github.com/apache/superset/pull/21065/files#diff-6a8138bf7b3a4c95d31569dfe4497ceb65ae07faddc47fb944a611ec7fb19776R86

Reply:
If you look at the line above that for useSelector, could you follow that method instead and import useState?

Reply:
But that test and file still imports useState and uses it instead of doing `React.useState`

Reply:
Indeed but the usage of the spied hook is different which might be affecting why these tests cannot do it that way, either way I'll give a spin to see if it works that way, if not possible we need to decide whether we move forward with it as is or not 🤔 I'll keep you posted @AAfghahi , thanks for the input 🎉